### PR TITLE
rgw: Added new counters for monitoring http status

### PR DESCRIPTION
**Pull request by prashant-kr312**
Signed-off-by: Prashant Kumar prashant.kr@flipkart.com

### CODE REVIEW

**Comment by prashant-kr312** [comment]
We are capturing req, failed_req, get and put requests from every RGW node. It would be nice to have continuously increasing HTTP counters for HTTP status. This gives more insight into RGW and should help in serviceability enhancement.
Tracker issue: http://tracker.ceph.com/issues/16971

**Comment by oritwas** [comment]
@prashant-kr312 , please rebase

File: src/rgw/librgw.cc [hunk 1]
```diff
@@ -293,6 +293,25 @@ namespace rgw {
     int http_ret = s->err.http_ret;
 
     req->log_format(s, "http status=%d", http_ret);
+    switch (http_ret / 100) {
```
Comment (amitkumar50) [inline src/rgw/librgw.cc]: @prashant-kr312 can't we use a dictionary that maps tag names to a tag handler to replace traditional case statements..

File: src/rgw/rgw_process.cc [hunk 2]
```diff
@@ -199,6 +199,25 @@ int process_request(RGWRados* store, RGWREST* rest, RGWRequest* req,
 
   req->log_format(s, "op status=%d", op_ret);
   req->log_format(s, "http status=%d", http_ret);
+  switch (http_ret / 100) {
```
Comment (amitkumar50) [inline src/rgw/rgw_process.cc]: @prashant-kr312 same

File: src/rgw/librgw.cc (hunk 1 above)
Comment (amitkumar50) [inline src/rgw/librgw.cc]: @prashant-kr312 Can you prefix Commit title with "rgw".
>It is required to prefix it with the subsystem or module you are changing.
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

**Comment by mattbenjamin** [comment]
@cbodley is there a problem with adding more perfcounters?  seems potentially useful

**Comment by cbodley** [comment]
@mattbenjamin nope, this looks useful

File: src/rgw/rgw_process.cc [hunk 3]
```diff
@@ -199,6 +199,25 @@ int process_request(RGWRados* store, RGWREST* rest, RGWRequest* req,
...
+      break;
+    case 4:
+      perfcounter->inc(l_rgw_http_4xx_count);
+      break;
+    case 5:
+      perfcounter->inc(l_rgw_http_5xx_count);
+      break;
+    default:
+      perfcounter->inc(l_rgw_http_NULL_count);
+  }
```
Comment (cbodley) [inline src/rgw/rgw_process.cc]: this could be a helper function in rgw_rest.cc so it doesn't get duplicated in librgw.cc

**Pull request closed without merging**